components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/DefaultLoginStorageDelegate.kt

```diff
@@ -0,0 +1,146 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.sync.logins
+
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import mozilla.components.concept.storage.Login
+import mozilla.components.concept.storage.LoginStorageDelegate
+import mozilla.components.lib.dataprotect.SecureAbove22Preferences
+
+/**
+ * A type of persistence operation, either 'create' or 'update'.
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+enum class Operation { CREATE, UPDATE }
+
+internal const val PASSWORDS_KEY = "passwords"
+
+/**
+ * [LoginStorage.Delegate] implementation.
+ *
+ * An abstraction that handles the persistence and retrieval of [LoginEntry]s so that Gecko doesn't
+ * have to.
+ *
+ * In order to use this class, attach it to the active [GeckoRuntime] as its `loginStorageDelegate`.
+ * It is not designed to work with other engines.
+ */
+class DefaultLoginStorageDelegate(
```

Comment:
Maybe call it `GeckoLoginStorageDelegate`, since it's "not designed to work with other engines"?